In [29]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [44]:
df=pd.read_csv("../data/techcrunch.csv")

In [45]:
df.head()

,permalink,company,numEmps,category,city,state,fundedDate,raisedAmt,raisedCurrency,round
0,lifelock,LifeLock,NaN,web,Tempe,AZ,1-May-07,6850000,USD,b
1,lifelock,LifeLock,NaN,web,Tempe,AZ,1-Oct-06,6000000,USD,a
2,lifelock,LifeLock,NaN,web,Tempe,AZ,1-Jan-08,25000000,USD,c
3,mycityfaces,MyCityFaces,7.0,web,Scottsdale,AZ,1-Jan-08,50000,USD,seed
4,flypaper,Flypaper,NaN,web,Phoenix,AZ,1-Feb-08,3000000,USD,a


In [46]:
df.shape

(1460, 10)

In [47]:
df.dtypes

permalink             str
company               str
numEmps           float64
category              str
city                  str
state                 str
fundedDate            str
raisedAmt           int64
raisedCurrency        str
round                 str
dtype: object

In [48]:
df.isna().mean().sort_values(ascending=False)*100

numEmps           61.164384
category           1.643836
city               1.232877
permalink          0.000000
company            0.000000
state              0.000000
fundedDate         0.000000
raisedAmt          0.000000
raisedCurrency     0.000000
round              0.000000
dtype: float64

# *Vu la proportion de données manquantes dans la colonne numEmps (soit 61.2%), il est préférale de l'omettre dans notre analyse*
# *Pour la colonne category on remplace les données manquantes par* ~ unknown et on laisse colonne city avec ses 1.2% de NA 

In [49]:
#Suppression de la colonne numEmp 
df.drop(columns=['numEmps'], errors='ignore', inplace=True)

In [50]:
#remplissage des valeurs manquantes de la colonne category avec la valeur "Unknown"
df['category'] = df['category'].fillna('unknown') 

In [51]:
df.columns

Index(['permalink', 'company', 'category', 'city', 'state', 'fundedDate',
       'raisedAmt', 'raisedCurrency', 'round'],
      dtype='str')

In [52]:
df['fundedDate'] = pd.to_datetime(df['fundedDate'], format='%d-%b-%y')

In [54]:
df.isna().sum()

permalink          0
company            0
category           0
city              18
state              0
fundedDate         0
raisedAmt          0
raisedCurrency     0
round              0
dtype: int64

In [55]:
df.dtypes

permalink                    str
company                      str
category                     str
city                         str
state                        str
fundedDate        datetime64[us]
raisedAmt                  int64
raisedCurrency               str
round                        str
dtype: object

In [56]:
df['fundedDate'].dt.year.agg(['min', 'max'])

min    1993
max    2008
Name: fundedDate, dtype: int32

In [57]:
df['fundedDate'].dt.year.value_counts().sort_index()

fundedDate
1993      1
1999     14
2000     17
2001     13
2002     12
2003     27
2004     60
2005    128
2006    253
2007    538
2008    397
Name: count, dtype: int64